### PR TITLE
[4.0] Cassiopeia - Add selector to list overflow style

### PR DESCRIPTION
**Pull request by YatharthVyas**
Pull Request for Issue #33617

### Summary of Changes
The `overflow: hidden` style applied to `ul` and `ol` in #33560 affected the navbar dropdowns which I hadn't tested properly. To rectify my mistake, I've added a selector that only applies the style to lists that are inside the article component

### Testing Instructions
1. `npm run build:css`
2. Check that all dropdowns are functional in the frontend
3. Visit 'Your Modules' blog in frontend and ensure that the lists are  properly indented

### Actual result BEFORE applying this Pull Request
Dropdowns don't work

### Expected result AFTER applying this Pull Request
Dropdowns work

image

### Documentation Changes Required
None

### CODE REVIEW

**Comment by richard67** [comment]
I have tested this item :white_check_mark: successfully on b1e7f290a81d79117e203a89cfb79af290f7ff29<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/33624">issues.joomla.org/tracker/joomla-cms/33624</a>.</sub>

**Comment by ghost** [comment]
I have tested this item :white_check_mark: successfully on b1e7f290a81d79117e203a89cfb79af290f7ff29<br /><br />Thanks all for solving the issue.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/33624">issues.joomla.org/tracker/joomla-cms/33624</a>.</sub>

**Comment by richard67** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/33624">issues.joomla.org/tracker/joomla-cms/33624</a>.</sub>

**Comment by wilsonge** [comment]
Thanks!

**Comment by wilsonge** [comment]
As a followup it would be good if someone could split the com_content specific styling from this globals file and create a dedicated scss file for it in a pages subfolder similar to the atum structure https://github.com/joomla/joomla-cms/blob/b1e7f290a81d79117e203a89cfb79af290f7ff29/administrator/templates/atum/scss/pages/_com_content.scss

**Comment by richard67** [comment]
> As a followup it would be good if someone could split the com_content specific styling from this globals file and create a dedicate scss file for it in a pages subfolder similar to the atum structure https://github.com/joomla/joomla-cms/blob/b1e7f290a81d79117e203a89cfb79af290f7ff29/administrator/templates/atum/scss/pages/_com_content.scss

@YatharthVyas Would you like to be that "someone"? 😄

**Comment by YatharthVyas** [comment]
> @YatharthVyas Would you like to be that "someone"? 😄

Sure, Thanks!

**Comment by YatharthVyas** [comment]
I noticed that the atum template's scss/pages/ imports have a message:
https://github.com/joomla/joomla-cms/blob/65d89d09721230b2e72945e6a94b82243b1cb40b/administrator/templates/atum/scss/template.scss#L59
which was added in commit https://github.com/joomla/joomla-cms/pull/26077/commits/482aea28a55c5a31b6443aabd9ee74fb62b9570e

I read the PR's discussion and couldn't figure out the reason behind this. This applies to the scss refactoring for Cassiopeia (discussed above) and I want to know what's the correct implementation for these import(s).

**Comment by richard67** [comment]
> 
> 
> I noticed that the atum template's scss/pages/ imports have a message:
> 
> https://github.com/joomla/joomla-cms/blob/65d89d09721230b2e72945e6a94b82243b1cb40b/administrator/templates/atum/scss/template.scss#L59
> 
> which was added in commit [482aea2](https://github.com/joomla/joomla-cms/commit/482aea28a55c5a31b6443aabd9ee74fb62b9570e)
> 
> I read the PR's discussion and couldn't figure out the reason behind this. This applies to the scss refactoring for Cassiopeia (discussed above) and I want to know what's the correct implementation for these import(s).

@dgrammatiko Can you answer that?

**Comment by dgrammatiko** [comment]
https://github.com/joomla/joomla-cms/blob/65d89d09721230b2e72945e6a94b82243b1cb40b/administrator/templates/atum/scss/template.scss#L59-L71

The problem -that most people don't even see in the above lines- is that we're creating a monolith that is totally inflexible. Of course, most people don't realise the impact of packaging all the css in a file because most of the times they are not even using Joomla correctly (eg extensive use of overrides). So, for those that will never override anything the monolith approach is fine, although a practice of the past that is really against today's needs. BUT if you ever want to override things due to the fact the routes/pages are already styled in the template you're in big trouble and either you have to completely change all the classes (and maybe the structure) of the produced HTML or you have to use `!important` everywhere. Also, the whole approach is actually fighting Joomla's architecture which is totally modular...

In short, routes (eg component views) should have their own css files. The same goes for modules and everything else that is outputting HTML... We're destroying what should have being a USP for the project (modularity) for the sake of using well establish (and totally outdated) patterns.
My 2c

**Comment by YatharthVyas** [comment]
I absolutely agree with your point

> In short, routes (eg component views) should have their own CSS files. 

In light of this, I am working on a PR that works on the page related styles for Cassiopeia (I'll later move on to Atum) and this is what I've done:
- Move the scss files for all component's to their respective build/media_source folder
- Import them in their respective component via Web Asset Manager
<br>

I have 2 questions for @dgrammatiko :
1. Certain styles are mapped to template variables like `$cassiopiea-gutter-width`. Should these be hardcoded now as we are separating the component CSS from the Cassiopeia template? 
2. Moving these styles to their component folder might not cause any effect for Cassiopeia but they will offset styling for 3rd Party Templates so is it worth it?

**Comment by dgrammatiko** [comment]
> Moving these styles to their component folder might not cause any effect for Cassiopeia but they will offset styling for 3rd Party Templates

Not really, the views will still have the default css so, 3rd pd could override the css by adding the relative file in their template css folder. Also you shouldn't really worry about 3rd pd templates as these will be brand new as the ones from version 3 hardly work with the bootstrap 5 markup 😉
